richtext-ui/src/commonMain/kotlin/com/halilibo/richtext/ui/string/Text.kt

```diff
@@ -177,14 +175,43 @@ private fun rememberAnimatedText(
               }
               animations[phraseIndex] = TextAnimation(phraseIndex, value)
             }
-            sharedAnimationState.value = sharedAnimationState.value.removeAnimation()
             animations.remove(phraseIndex)
           }
         }
       if (phrases.isComplete) {
         textToRender.value = phrases.annotatedString
       }
     }
+    LaunchedEffect(annotated) {
+      debouncedTextFlow.value = annotated
+      // If we detect a new phrase, kick off the animation now.
+      val phrases = annotated.segmentIntoPhrases(renderOptions, isComplete = !isLeafText)
+      if (phrases.hasNewPhrasesFrom(readyToAnimateText.value)) {
+        if (phrases != readyToAnimateText.value) {
+          readyToAnimateText.value = phrases
+          animationUpdate()
+        }
+      }
+    }
+    LaunchedEffect(isLeafText, annotated) {
+      if (!isLeafText) {
```

Comment:
nit: for this and the block below, I prefer guard statements that un-nest the rest of the block

`if (isLeafText) return@LaunchedEffect`